eval.c

```diff
@@ -527,20 +546,21 @@ DO_OP_RETURN: {
 	retval = POP();
 
 	/* close upvalues */
-	if (state->upvalues_idx != 0) {
-		for (i = state->upvalues_idx - 1; i >= 0; i -= 1) {
-			uv = state->upvalues[i];
-			if (!uv || !uv->is_open || uv->v.idx < frame->bp)
+	if (cb_vm_state.upvalues_idx != 0) {
```

Comment:
Do we really need this global list of upvalues anymore? It might be enough to just close the upvalues on the function that's returning

Reply:
yup because a function's upvalues list is just the ones that are bound to that function; a function would need a second list for the ones that refer to values in that function